vllm/engine/arg_utils.py

```diff
@@ -484,6 +488,13 @@ def add_cli_args(
             help=
             'The name of the draft model to be used in speculative decoding.')
 
+        parser.add_argument(
+            '--extra-inputs-for-draft-model',
+            type=nullable_str,
+            default=EngineArgs.extra_inputs_for_draft_model,
+            help='Extra model inputs used by draft model.'
+            'These should come as outputs from the target model.')
```

Comment:
Can you provide an example of how this is used?

Reply:
It was used for notifying the worker that draft model takes in additional inputs e.g., for medusa, you'd need to pass "hidden_states" as the value in this argument. But, I think this can be removed. I will think of a better way which doesn't need additional command line args.

One way I am thinking is that we add an optional input_config attribute to HF configs (e.g., in `MedusaConfig`, `input_config = {"hidden_states": (hidden_states_shape, hidden_states_dtype)}`). This way, we can use just this attribute to handle the inputs and outputs of draft and target mode resp.

As you suggested, I'll create a separate PR for this which focuses only on handling the custom inputs and outputs of the models.

Reply:
This is removed now. We use model's hf_config to infer what additional inputs are required.